contracts/interchain-token-service/tests/deploy_remote_interchain_token.rs

```diff
@@ -0,0 +1,196 @@
+use soroban_sdk::Bytes;
+use soroban_sdk::IntoVal;
+use soroban_sdk::Symbol;
+use soroban_sdk::{testutils::Address as _, Address, BytesN, String};
+use soroban_token_sdk::metadata::TokenMetadata;
+use utils::setup_env;
+use utils::setup_gas_token;
+use utils::TokenMetadataExt;
+
+#[test]
+fn deploy_remote_interchain_token_succeeds() {
+    let (env, client, _, _, _) = setup_env();
+
+    let sender = Address::generate(&env);
+    let gas_token = setup_gas_token(&env, &sender);
+    let minter: Option<Address> = None;
+    let salt = BytesN::<32>::from_array(&env, &[1; 32]);
+    let token_metadata = TokenMetadata::new(&env, "name", "symbol", 6);
+    let initial_supply = 1;
+
+    let token_id = client.mock_all_auths().deploy_interchain_token(
+        &sender,
+        &salt,
+        &token_metadata,
+        &initial_supply,
+        &minter,
+    );
+
+    let destination_chain = String::from_str(&env, "ethereum");
+    client
+        .mock_all_auths()
+        .set_trusted_chain(&destination_chain);
+
+    let deployed_token_id = client.mock_all_auths().deploy_remote_interchain_token(
+        &sender,
+        &salt,
+        &destination_chain,
+        &gas_token,
+    );
+    assert_eq!(token_id, deployed_token_id);
+
+    goldie::assert!(events::fmt_emitted_event_at_idx::<
+        InterchainTokenDeploymentStartedEvent,
+    >(&env, -4));
+}
+
+#[test]
+fn deploy_remote_interchain_token_auth_test() {
+    let (env, client, _, gas_service, _) = setup_env();
+
+    let sender = Address::generate(&env);
+    let gas_token = setup_gas_token(&env, &sender);
+    let minter: Option<Address> = None;
+    let salt = BytesN::<32>::from_array(&env, &[1; 32]);
+    let token_metadata = TokenMetadata::new(&env, "name", "symbol", 6);
+    let initial_supply = 1;
+
+    let token_id = client.mock_all_auths().deploy_interchain_token(
+        &sender,
+        &salt,
+        &token_metadata,
+        &initial_supply,
+        &minter,
+    );
+
+    let destination_chain = String::from_str(&env, "ethereum");
+    let its_hub_chain = String::from_str(&env, "axelar");
+    let its_hub_address = String::from_str(&env, "its_hub_address");
+
+    client
+        .mock_all_auths()
+        .set_trusted_chain(&destination_chain);
+
+    let deployed_token_id = client.mock_all_auths().deploy_remote_interchain_token(
+        &sender,
+        &salt,
+        &destination_chain,
+        &gas_token,
+    );
+    assert_eq!(token_id, deployed_token_id);
+
+    let message = Message::DeployInterchainToken(DeployInterchainToken {
+        token_id,
+        name: token_metadata.name.clone(),
+        symbol: token_metadata.symbol.clone(),
+        decimals: token_metadata.decimal as u8,
+        minter: None,
+    });
+
+    let payload = HubMessage::SendToHub {
+        destination_chain: destination_chain.clone(),
+        message,
+    }
+    .abi_encode(&env);
+
+    let transfer_auth = auth_invocation!(&env,
```

Comment:
Can you make the usage for this simpler, similar to assert_invoke_auth_ok so you just need to pass in the function call as is? This way the function call is typed by using a client

Also after this change, the succeeds and auth check shouldn't be separate tests, auth should be tested as part of success

Reply:
Created a ticket to track this, and will be working on it: https://axelarnetwork.atlassian.net/browse/AXE-6986